src/routes/api/api-router.ts

```diff
@@ -20,6 +20,36 @@ ApiRouter.use(urlencoded({ extended: false }));
 ApiRouter.use(json());
 ApiRouter.use(LogMiddleware);
 
+// Verify SLAuth headers to make sure that no open access was allowed for these endpoints
+// And also log how the request was authenticated
+
+ApiRouter.use(
+	async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+		const mechanism = req.get("X-Slauth-Mechanism");
+		const issuer = req.get("X-Slauth-Issuer");
+		const principal = req.get("X-Slauth-Principal");
+
+		if(!mechanism || mechanism === "open") {
+			res.status(401).json({error: "Open access not allowed"});
+			return;
```

Comment:
Should we maybe log this as a warn when someone's trying to authenticate without correct credentials?

Reply:
Makes sense, will do

Reply:
Also, I'm not sure I'm a fan of this conditional.  When you're doing a security check, you check for what you expect to continue, not check for what _not_ to continue.

Shouldn't we also check the other headers? is there something atlassian specific in the issues, principal, etc?  Shouldn't we also check the usergroup to make sure that whoever they are in atlassian has access to this API?

Reply:
@KAbakumovAtl :arrow_up:

Reply:
No, it is the responsibility of POCO